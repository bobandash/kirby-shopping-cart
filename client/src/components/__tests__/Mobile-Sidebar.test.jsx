import {describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from "@testing-library/react";
import MobileSideBar from '../Mobile-Sidebar';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

// setups props
describe('Footer', () => {
  describe('DOM', () => {
    it("matches snapshot", () => {
      const routerWrapper = ({children}) => <BrowserRouter>{children}</BrowserRouter>
      const {container} = render(<MobileSideBar/>, {wrapper: routerWrapper});
      expect(container).toMatchSnapshot();
    })
  })

  // note: the rest of the links don't do anything
  describe('Footer Links', () => {
    beforeEach(() => {
      const routerWrapper = ({children}) => <BrowserRouter>{children}</BrowserRouter>
      render(<MobileSideBar />, {wrapper: routerWrapper});      
    })

    it('Shop All works', async () => {
      const link = screen.getByRole("link", {name: "Shop All"});
      expect(link).toBeInTheDocument();
      const user = userEvent.setup();
      await user.click(link);
      expect(window.location.pathname).toBe('/category/all');
    })

    it('Games works', async () => {
      const link = screen.getByRole("link", {name: "Games"});
      expect(link).toBeInTheDocument();
      const user = userEvent.setup();
      await user.click(link);
      expect(window.location.pathname).toBe('/category/games');
    })
    
    it('Plushies works', async () => {
      const link = screen.getByRole("link", {name: "Plushies"});
      expect(link).toBeInTheDocument();
      const user = userEvent.setup();
      await user.click(link);
      expect(window.location.pathname).toBe('/category/plushies');
    })

    it('Keychains works', async () => {
      const link = screen.getByRole("link", {name: "Keychains"});
      expect(link).toBeInTheDocument();
      const user = userEvent.setup();
      await user.click(link);
      expect(window.location.pathname).toBe('/category/keychains');
    }) 
  })
})