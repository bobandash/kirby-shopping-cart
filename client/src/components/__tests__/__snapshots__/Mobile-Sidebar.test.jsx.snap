// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > DOM > matches snapshot 1`] = `
<div>
  <aside
    class="_mobile-sidebar_ff7a54 _hidden_ff7a54"
  >
    <ul
      class="_sidebar-links_ff7a54"
    >
      <li
        class="_link_ff7a54 _hidden_ff7a54"
      >
        <a
          href="/category/all"
        >
          Shop All
        </a>
      </li>
      <li
        class="_link_ff7a54 _hidden_ff7a54"
      >
        <a
          href="/category/games"
        >
          Games
        </a>
      </li>
      <li
        class="_link_ff7a54 _hidden_ff7a54"
      >
        <a
          href="/category/plushies"
        >
          Plushies
        </a>
      </li>
      <li
        class="_link_ff7a54 _hidden_ff7a54"
      >
        <a
          href="/category/keychains"
        >
          Keychains
        </a>
      </li>
    </ul>
  </aside>
</div>
`;
